packages/std/src/api.rs

```diff
@@ -3,32 +3,16 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-use crate::errors::{StdError, SystemError};
+use crate::errors::StdError;
 use crate::HumanAddr;
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub enum ApiResult<T, E: std::error::Error = ApiError> {
-    Ok(T),
-    Err(E),
-}
+pub type ApiResult<T> = Result<T, ApiError>;
```

Comment:
Okay, this is great if it works.

I saw this work with `to_vec` and `JsonSchema`, but a bit confused as I didn't think `std::result::Result` new about either of those two libraries (serde or schemars). Any idea how this works?

Reply:
Good question. Turns out this is implemented here: https://github.com/GREsau/schemars/blob/v0.7.1/schemars/src/json_schema_impls/core.rs

Reply:
Nice one. Inspired by that I found the serde implementations:

https://github.com/serde-rs/serde/blob/master/serde/src/ser/impls.rs#L574-L590

Reply:
So all this mental gymnastics was totally unnecessary.
Happy you are simplifying this :smile:

Reply:
I think the real issue is that we cannot derive from Result, making it our own type and adding custom functionality. This is a Rust limitation that seems to come up in multiple places. In general, implementing Into/From is the better option. But here we gain a lot more from using the standard library.